bio-qiime-pipeline/src/utils.py

```diff
@@ -145,3 +147,79 @@ def getMetadata(sample: CustomSample, metadataFileNme: str) -> Path:
         metadataPath = metadataPath.parent / f"{metadataPath.stem}.tsv"
 
     return metadataPath
+
+
+def getDatasetName(experiment: Experiment[CustomDataset], step: int) -> str:
+    if step < 1 or step > 5:
+        raise ValueError(">> [Microbiome analysis] Step number has to be between 1 and 5")
+
+    if experiment.parameters["barcodeColumn"]:
+        prefix = f"{experiment.id} - Step 1: Demux"
+    else:
+        prefix = f"{experiment.id} - Step 1: Import"
+
+    paramList = [
+        str(experiment.dataset.id),
+        str(experiment.parameters["metadataFileName"]),
+        str(experiment.parameters["barcodeColumn"]),
+        str(experiment.parameters["forwardAdapter"]),
+        str(experiment.parameters["reverseAdapter"])
+    ]
+
+    if step > 1:
+        prefix = f"{experiment.id} - Step 2: Denoise"
+        paramList.extend([
+            str(experiment.parameters["trimLeftF"]),
+            str(experiment.parameters["trimLeftR"]),
+            str(experiment.parameters["truncLenF"]),
+            str(experiment.parameters["truncLenR"]),
+        ])
+
+    if step == 3:
+        prefix = f"{experiment.id} - Step 3: Phylogenetic tree"
+        paramList.append("Step 3")
+
+    if step == 4:
+        prefix = f"{experiment.id} - Step 4: Alpha & Beta diversity"
+        paramList.extend([
+            str(experiment.parameters["samplingDepth"]),
+            str(experiment.parameters["maxDepth"]),
+            str(experiment.parameters["targetTypeColumn"])
+        ])
+
+    if step == 5:
+        prefix = f"{experiment.id} - Step 5: Taxonomic analysis"
+        paramList.append(experiment.parameters["classifier"])
+
+    if not experiment.parameters["useCache"]:
+        return prefix
+
+    return hashCacheName(prefix, "_".join(paramList))
+
+
+def getCaches(experiment: Experiment[CustomDataset]) -> Dict[int, CustomDataset]:
```

Comment:
Change this function so it accepts only a dataset name and it returns a cached dataset if it found one, otherwise it returns None.